### PR TITLE
Add new settings for DatePicker labels background and foreground whil…

### PR DESCRIPTION
**Pull request by YlaSplinter**
…e hovering

This pull request attempts to provide a a new feature which allows to customise background and foreground of DatePicker labels while hovering.
 - commit a815f5f contains changes to:
      - `DatePickerSettings` class to add two new DateArea settings
      - `CalendarPanel` to change behaviour of hover event and change to default state methods

### CODE REVIEW

File: Project/src/main/java/com/github/lgooddatepicker/components/DatePickerSettings.java [hunk 1]
```diff
@@ -78,6 +79,7 @@
         CalendarTextWeekdays(Color.black),
         CalendarTextWeekNumbers(Color.black),
         TextClearLabel(new JLabel().getForeground()),
+        TextHoverLabel(Color.black),
```
Comment (WiseEagleOwl) [inline Project/src/main/java/com/github/lgooddatepicker/components/DatePickerSettings.java:82]: This color never used in the rest of the Code, I think something is missing in your code here.

File: Project/src/main/java/com/github/lgooddatepicker/components/CalendarPanel.java [hunk 2]
```diff
@@ -940,7 +940,8 @@ private void labelIndicatorMouseEntered(MouseEvent e) {
             return;
         }
         // Highlight the label.
-        label.setBackground(new Color(184, 207, 229));
+        label.setBackground(settings.getColor(DateArea.BackgroundHoverLabel));
+        label.setForeground(settings.getColor(DateArea.TextHoverLabel));
```
Comment (WiseEagleOwl) [inline Project/src/main/java/com/github/lgooddatepicker/components/CalendarPanel.java:944]: Why is the *Foreground* set here additionally, if only the *Background* color is now customizeable? Is something missing in the code or am I missing something?

File: Project/src/main/java/com/github/lgooddatepicker/components/CalendarPanel.java [hunk 3]
```diff
@@ -964,13 +965,16 @@ private void labelIndicatorSetColorsToDefaultState(JLabel label) {
...
+            label.setForeground(settings.getColor(DateArea.TextMonthAndYearMenuLabels));
             monthAndYearInnerPanel.setBackground(settings.getColor(DateArea.BackgroundMonthAndYearMenuLabels));
         }
         if (label == labelSetDateToToday) {
             label.setBackground(settings.getColor(DateArea.BackgroundTodayLabel));
+            label.setForeground(settings.getColor(DateArea.TextTodayLabel));
         }
         if (label == labelClearDate) {
             label.setBackground(settings.getColor(DateArea.BackgroundClearLabel));
+            label.setForeground(settings.getColor(DateArea.TextClearLabel));
```
Comment (WiseEagleOwl) [inline Project/src/main/java/com/github/lgooddatepicker/components/CalendarPanel.java:977]: Same question as above, the Background is customizeable but you are setting the *Foreground* here :thinking:

File: Project/src/main/java/com/github/lgooddatepicker/components/DatePickerSettings.java (hunk 1 above)
Comment (WiseEagleOwl) [reply Project/src/main/java/com/github/lgooddatepicker/components/DatePickerSettings.java:82]: Oh sorry this was my mistake. I got confused by the `setForeground` terminology, the foreground is the color of the text. and is actually used. Please disregard this comment.

File: Project/src/main/java/com/github/lgooddatepicker/components/CalendarPanel.java (hunk 2 above)
Comment (WiseEagleOwl) [reply Project/src/main/java/com/github/lgooddatepicker/components/CalendarPanel.java:944]: Please disregard my precious comment `setForground` means setting the text color in this context, which is correct.

File: Project/src/main/java/com/github/lgooddatepicker/components/CalendarPanel.java (hunk 3 above)
Comment (WiseEagleOwl) [reply Project/src/main/java/com/github/lgooddatepicker/components/CalendarPanel.java:977]: Please disregard my precious comment. `setForground` means setting the text color in this context, which is needed here to restore the color after highlighting is finished.

**Comment by YlaSplinter** [comment]
I will change permissions tonight.

**Comment by WiseEagleOwl** [comment]
I still can't modify your branch :slightly_frowning_face: . In case you have trouble finding the setting, take a look here:
https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests

File: Project/src/main/java/com/github/lgooddatepicker/components/DatePickerSettings.java [hunk 4]
```diff
@@ -63,6 +63,7 @@
      */
     public enum DateArea {
         BackgroundClearLabel(new Color(240, 240, 240)),
+        BackgroundHoverLabel(new Color(184, 207, 229)),
```
[the remark this answers is not in the document]
Comment (WiseEagleOwl) [reply Project/src/main/java/com/github/lgooddatepicker/components/DatePickerSettings.java]: done with 88d23bf

File: Project/src/main/java/com/github/lgooddatepicker/components/DatePickerSettings.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (WiseEagleOwl) [reply Project/src/main/java/com/github/lgooddatepicker/components/DatePickerSettings.java]: default color fix in 6293fe0

**Comment by WiseEagleOwl** [comment]
@YlaSplinter 
Thank you for contribution to *LGoodDatePicker*!

**Pull request merged**